src/components/pages/poll-page/components/answers-sheet/AnswersSheet.tsx

```diff
@@ -64,213 +37,216 @@ export const getProgress = (answers: Answer[], questions: Question[]) => {
-                            { value: '0', label: 'ні' },
-                          ]}
-                          sx={{
-                            display: 'flex',
-                            justifyContent: 'flex-start',
-                            gap: '36px',
-                          }} //TODO remove inline styles when refactor
-                          name={question.id}
-                        />
-                      ) : (
-                        <TextArea
-                          rowsNumber={numberRowsTextArea}
-                          sx={sxStyles.textArea}
-                          name={question.id}
-                        />
-                      )}
-
-                      {question.criteria && (
-                        <p className={styles['question-criteria']}>
-                          {question.criteria}
-                        </p>
-                      )}
-                    </div>
-                  ))}
-                  <Button
-                    className={styles['button']}
-                    text={
-                      isTheLast ? 'Завершити опитування' : 'Наступні питання'
-                    }
-                    type="submit"
-                    disabled={isTheLast && !isValid}
-                    onClick={async () => {
-                      answer(values);
-                      if (!isTheLast) {
-                        setCurrent(prev => ++prev);
-                        window.scrollTo({ top: 0, behavior: 'smooth' });
-                      } else {
-                        setIsSendingStatus(SendingStatus.LOADING);
-                        try {
-                          for (let i = 0; i < answers.length; i++) {
-                            if (i == answers.length - 1) {
-                              answers[i].value = answers[i].value
-                                .toString()
-                                .trim();
-                            }
-                            if (answers[i].value.length === 0) {
-                              answers = answers.filter(
-                                item => item !== answers[i],
-                              );
-                            }
-                            answers[i].value = answers[i].value.toString();
-                          }
-                          await PollAPI.createTeacherGrade(
-                            { answers },
-                            disciplineTeacherId,
-                          );
-                          setIsSendingStatus(SendingStatus.SUCCESS);
-                        } catch (error) {
-                          displayError(error);
-                          setIsSendingStatus(SendingStatus.ERROR);
-                        }
+                    className={styles['form']}
+                  >
+                    {currentQuestions?.questions.map((question, key) => (
+                      <SingleQuestion
+                        key={key}
```

Comment:
```suggestion
                        key={question.id}
```
never use an index as s key

Reply:
why? and look here 
![image](https://github.com/fictadvisor/fictadvisor-web/assets/68693183/89116f11-caa5-4bce-9126-60d75b72bd69)

Reply:
![image](https://github.com/fictadvisor/fictadvisor-web/assets/86970828/d59d5883-33f4-432d-998a-b6a25c1b4b39)
[docs](https://legacy.reactjs.org/docs/lists-and-keys.html)